src/main/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidator.java

```diff
@@ -610,6 +592,38 @@ private void validatePresenceOfMultipleYearFilerFields(IntangibleAssetsResource
         }
     }
 
+    private boolean validateSubResourceCost(IntangibleAssetsResource resource, String period, Errors errors, IntangibleSubResource intangibleSubResource, boolean isInvalid) {
+
+        if (resource.getCost() != null) {
+            Long periods = period.equals(COST_AT_PERIOD_START) ? resource.getCost().getAtPeriodStart() : resource.getCost().getAtPeriodEnd();
+            if(periods == null) {
+                addError(errors, valueRequired, getJsonPath(intangibleSubResource, period));
+                isInvalid = true;
+            }
+        }
+        else {
+
+            isInvalid = true;
```

Comment:
I don't think the sub resource is invalid if cost is null

Reply:
Removed the else block from the code